src/Simulation/Simulators.Tests/StackTraceTests.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Xunit;
+
+using System;
+using System.Threading.Tasks;
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Simulators.Tests.Circuits;
+using Microsoft.Quantum.Simulation.Simulators.Exceptions;
+using Xunit.Abstractions;
+using System.Text;
+
+namespace Microsoft.Quantum.Simulation.Simulators.Tests
+{
+    public class StackTraceTests
+    {
+        const string namespacePrefix = "Microsoft.Quantum.Simulation.Simulators.Tests.Circuits.";
+
+        private readonly ITestOutputHelper output;
+        public StackTraceTests(ITestOutputHelper output)
+        {
+            this.output = output;
+        }
+
+        [Fact]
+        public void AlwaysFail4Test()
+        {
+            ToffoliSimulator sim = new ToffoliSimulator();
```

Comment:
maybe we should try it with multiple simulators?

Reply:
Here for the test I just need a class implementing `SimulatorBase` and no simulator specific logic is involved.